frontend/src/app/Catalog/CatalogView/CatalogView.tsx

```diff
@@ -85,8 +133,64 @@ const CatalogView = (props: CatalogViewProps) => {
 			setOpen(true);
 		}
 	}, [coursePreview, data]);
+	
+	
+	useEffect(() => {
+		if (course !== null) {	
+			if (false && legacyId !== null) {		
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+				// let temp = semester.split(' ');
+				// dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+				// axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+				// 	dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+				// 	});	
+			} else {
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+			}
+		}
+	}, [course?.courseNumber]);
 
-	const [playlists, sections] = useMemo(() => {
+	useEffect(() => {
+		legacyId = (store.getState() as any).enrollment?.context?.courses?.find(
+			(c: any) => c.abbreviation === abbreviation && c.course_number === courseNumber
+		)?.id ?? null
+		
+		if (legacyId !== null) {
+			let temp = semester.split(' ');
+			dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+			axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+				dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+				});
+		}
+		
+	}, [enrollContext])
+
+	useEffect(() => {
+		if (course !== null) {
```

Comment:
lets prefer `!course` and use `&&` to condense the if statements

Reply:
not sure how to do this either